pkg/user/tx_client_test.go

```diff
@@ -447,3 +451,358 @@ func setupEstimatorService(t *testing.T) *mockEstimatorServer {
 	t.Cleanup(mes.stop)
 	return mes
 }
+
+const (
+	BroadcastTestChainID = "test-broadcast-chain"
+	bufferSize           = 1024 * 1024
+)
+
+type BroadcastTestSuite struct {
+	suite.Suite
+	kr      keyring.Keyring
+	encCfg  encoding.Config
+	signer  *user.Signer
+	account string
+	accAddr sdk.AccAddress
+}
+
+func (s *BroadcastTestSuite) SetupSuite() {
+	s.encCfg = encoding.MakeConfig()
+	s.kr = keyring.NewInMemory(s.encCfg.Codec)
+	s.account = "test_broadcast_account"
+	info, _, err := s.kr.NewMnemonic(s.account, keyring.English, "", "", hd.Secp256k1)
+	s.Require().NoError(err)
+	pubKey, err := info.GetPubKey()
+	s.Require().NoError(err)
+	s.accAddr = sdk.AccAddress(pubKey.Address())
+
+	s.signer, err = user.NewSigner(s.kr, s.encCfg.TxConfig, BroadcastTestChainID, user.NewAccount(s.account, 0, 0))
+	s.Require().NoError(err)
+}
+
+func TestBroadcastTestSuite(t *testing.T) {
+	suite.Run(t, new(BroadcastTestSuite))
+}
```

Comment:
upon merging, imo we should write a flup issue that tries to address this test as this feels like a kind of insane amount of tech debt to test really simple functionality